core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -329,6 +339,39 @@ public synchronized TopLevelItem createProject( TopLevelItemDescriptor type, Str
     }
 
     /**
+     * Populate a {@link Categories} object from a specific {@link ItemGroup}.
+     *
+     * @return A object that represents a set of {@link Category}
+     */
+    @Restricted(NoExternalUse.class)
+    public static Categories getCategories(Authentication a, ItemGroup c) {
```

Comment:
Well it would seem simpler and clearer to inline it into `View`, but as I say so long as it is restricted it does not matter much.